### PR TITLE
Add dedicated methods to add/remove motor/contact points

### PR DESCRIPTION
**Pull request by duburcqa**

### CODE REVIEW

**Comment by duburcqa** [comment]
ping @matthieuvigne

File: core/src/Model.cc [hunk 1]
```diff
@@ -188,6 +193,262 @@ namespace jiminy
...
+
+            /* Make sure that no user-specified torque limits or motor inertias are used,
+               since it does not make sense to add motors otherwise */
+            if (!mdlOptions_->joints.torqueLimitFromUrdf || mdlOptions_->joints.enableMotorInertia)
+            {
+                std::cout << "Warning - Model::addMotors - Be careful, adding motors disable user-specified torque limits and motor inertia to prevent dimensional mismatch." << std::endl;
+                configHolder_t mdlOptions = getOptions();
+                configHolder_t & jointOptionsHolder = boost::get<configHolder_t>(mdlOptions.at("joints"));
+                boost::get<bool>(jointOptionsHolder.at("torqueLimitFromUrdf")) = true;
+                boost::get<vectorN_t>(jointOptionsHolder.at("torqueLimit")) = vectorN_t();
```
Comment (matthieuvigne) [inline core/src/Model.cc]: No need to empty this vector, simply disable the option.

File: core/src/Model.cc [hunk 2]
```diff
@@ -188,6 +193,262 @@ namespace jiminy
...
+                boost::get<bool>(jointOptionsHolder.at("enableMotorInertia")) = false;
+                boost::get<vectorN_t>(jointOptionsHolder.at("motorInertia")) = vectorN_t();
```
Comment (matthieuvigne) [inline core/src/Model.cc]: same

File: core/src/Model.cc [hunk 3]
```diff
@@ -188,6 +193,262 @@ namespace jiminy
...
+
+            /* Make sure that no user-specified torque limits or motor inertias are used,
+               since it does not make sense to add motors otherwise */
+            if (!mdlOptions_->joints.torqueLimitFromUrdf || mdlOptions_->joints.enableMotorInertia)
+            {
+                std::cout << "Warning - Model::removeMotors - Be careful, adding motors disable user-specified torque limits and motor inertia to prevent dimensional mismatch." << std::endl;
+                configHolder_t mdlOptions = getOptions();
+                configHolder_t & jointOptionsHolder = boost::get<configHolder_t>(mdlOptions.at("joints"));
+                boost::get<bool>(jointOptionsHolder.at("torqueLimitFromUrdf")) = true;
+                boost::get<vectorN_t>(jointOptionsHolder.at("torqueLimit")) = vectorN_t();
```
Comment (matthieuvigne) [inline core/src/Model.cc]: same

File: core/src/Model.cc [hunk 4]
```diff
@@ -188,6 +193,262 @@ namespace jiminy
...
+                boost::get<bool>(jointOptionsHolder.at("enableMotorInertia")) = false;
+                boost::get<vectorN_t>(jointOptionsHolder.at("motorInertia")) = vectorN_t();
```
Comment (matthieuvigne) [inline core/src/Model.cc]: same

File: core/src/Model.cc [hunk 5]
```diff
@@ -968,17 +1220,23 @@ namespace jiminy
...
             }
 
-            // Trigger biased model regeneration and info extraction if necessary
-            if (isFlexibleModelInvalid || isModelInvalid)
+            // Trigger biased model regeneration
+            if (isFlexibleModelInvalid || isCurrentModelInvalid)
             {
                 reset();
             }
+
```
Comment (matthieuvigne) [inline core/src/Model.cc]: remove

**Pull request merged**